Detectors/MUON/MCH/Raw/Decoder/src/UserLogicEndpointDecoder.h

```diff
@@ -119,6 +119,15 @@ size_t UserLogicEndpointDecoder<CHARGESUM, VERSION>::append(Payload buffer)
 
     int gbt = ulword.linkID;
 
+    // The User Logic uses the condition gbt=15 to identify special control and diagnostics words
+    // that are generated internally and do not contain data coming from the front-end electronics.
+    if (gbt == 15) {
+      // TODO: the exact format of the UL control words is still being defined and tested,
+      // hence proper decoding will be implemented once the format is finalized.
+      // For the moment we simply avoid throwing an exception when linkID is equal to 15
+      continue;
```

Comment:
I don't understand how that's going to prevent the exception below ? Did you mean `return 0;` instead of `continue` maybe ?

Anyway, I would put that == 15 test within the <0 || > 11 test itself to avoid a double if for regular gbt values (i.e. between 0 and 11) (assuming here an extra if has potential performance implications, but that should be measured of course)

Reply:
Unless I am really overlooking something, the `continue` statement prevents the code in lines 131 and below to be executed, the same way as we skip words that are zero or equal to `0xFEEDDEEDFEEDDEED`.

Also, the standard decoding (lines 141 and below) should not be applied to those words, since they do not contain front-end data. Therefore an `if (gbt == 15)` statement is needed anyhow.

One possibility to prevent the double `if` for each 64-bit word could be to replace the `if (gbt < 0 || gbt > 11)` statement on line 131 by `else if (gbt < 0 || gbt > 11)`. From the code flow point of view, that would result in something like
```
If gbt == 15 then 
    handle the special words
otherwise if invalid
    throw an exception or call the error handler
```

What do you think?

Reply:
My bad. Forgot that 1st comment, was looking too fast so I've missed the external loop altogether ;-). So the continue actually makes perfect sense. Sorry about that.

For the double if, yes an else would do it.